src/DocumentDB.ChangeFeedProcessor.IntegrationTests/V3CompatibilityTests.cs

```diff
@@ -0,0 +1,273 @@
+                (FeedProcessing.IChangeFeedObserverContext context, IReadOnlyList<Document> docs) =>
+                {
+                    foreach (Document doc in docs)
+                    {
+                        receivedIds.Add(int.Parse(doc.Id));
+                    }
+
+                    if (receivedIds.Count == 10)
+                    {
+                        firstSetOfResultsProcessed.Set();
+                    }
+
+                    if (receivedIds.Count == 20)
+                    {
+                        secondSetOfResultsProcessed.Set();
+                    }
+
+                    return Task.CompletedTask;
+                });
+
+            IChangeFeedProcessor changeFeedProcessorBuilder = await new ChangeFeedProcessorBuilder()
+                    .WithObserverFactory(observerFactory)
+                    .WithHostName("smoke_test")
+                    .WithFeedCollection(this.MonitoredCollectionInfo)
+                    .WithLeaseCollection(this.LeaseCollectionInfo)
+                    .BuildAsync();
+
+            await changeFeedProcessorBuilder.StartAsync();
+            await this.WaitUntilLeaseStoreIsInitializedAsync();
+
+            // Inserting some documents
+            using (DocumentClient client = new DocumentClient(this.MonitoredCollectionInfo.Uri, this.MonitoredCollectionInfo.MasterKey, this.MonitoredCollectionInfo.ConnectionPolicy))
+            {
+                Uri collectionUri = UriFactory.CreateDocumentCollectionUri(this.MonitoredCollectionInfo.DatabaseName, this.MonitoredCollectionInfo.CollectionName);
+
+                foreach (int id in expectedIds.Take(10))
+                {
+                    await client.CreateDocumentAsync(collectionUri, new { id = id.ToString() });
+                }
+            }
+
+            Assert.True(firstSetOfResultsProcessed.WaitOne(IntegrationTest.changeWaitTimeout), "Timed out waiting for first set of items to be received.");
+
+            await changeFeedProcessorBuilder.StopAsync();
+
+            // At this point we have leases for V2, so we will simulate V3 by manually adding LeaseToken and removing PartitionId
+            using (DocumentClient client = new DocumentClient(this.LeaseCollectionInfo.Uri, this.LeaseCollectionInfo.MasterKey, this.LeaseCollectionInfo.ConnectionPolicy))
+            {
+                Uri collectionUri = UriFactory.CreateDocumentCollectionUri(this.LeaseCollectionInfo.DatabaseName, this.LeaseCollectionInfo.CollectionName);
+
+                IDocumentQuery<JObject> query = client.CreateDocumentQuery<JObject>(collectionUri, "SELECT * FROM c").AsDocumentQuery();
+                while (query.HasMoreResults)
+                {
+                    foreach (JObject lease in await query.ExecuteNextAsync())
+                    {
+                        string leaseId = lease.Value<string>("id");
+                        if (leaseId.Contains(".info"))
```

Comment:
nit: we can rule out .info by WEHERE clause. OK though

Reply:
I put it there to detect that the file exists